### PR TITLE
#[UIC-1256]: Publish default filter during dashboad

### PR DESCRIPTION
**Pull request by ankursinghal2005**
[UIC-1256](https://guavus-jira.atlassian.net/browse/UIC-1256)

### CATEGORY

Choose one

- [ ] Bug Fix
- [x] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Publish filters with default value on the dashboard

### TEST PLAN

NA
### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [x] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 1]
```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
...
-    );
-  } catch (e) {
-    //
+
+  const getFiltersFromFilterConfig = (filter_configs) => {
+    const filters = {};
+    filter_configs.forEach( filter => {
+      const defaultValue = filter.hasOwnProperty("defaultValue")?filter["defaultValue"]:"";
+      var valArr = defaultValue!==""?defaultValue.split(";"):[]; 
+      if(valArr.length > 0){
```
Comment (jitendra-kumawat) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: rename it to values

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 2]
```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
...
+        filters[filter["column"]] = valArr;
+      }
+    })
+    return filters;
   }
```
Comment (jitendra-kumawat) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: 1.correct code formatting 
2. we should trim string before add empty check or better way to validate empty string
3. add column prop check

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 3]
```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
...
 
+  // update filter with filter_box Viz default values
+  dashboard.slices.forEach(slice => {
+     let defultFilters = {};
+
```
Comment (jitendra-kumawat) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: rename defultFilters  to defaultFilters

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 4]
```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
...
+      // As per the current support only filter_box can publish global default filters
+      if(slice.form_data.viz_type == "filter_box"){
+
```
Comment (jitendra-kumawat) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: can we pick filter_box value from const files

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 5]
```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
...
+        const publish_columns = slice.form_data.hasOwnProperty("publish_columns")?slice.form_data.publish_columns:[];
+        const filterConfigFilters = getFiltersFromFilterConfig(slice.form_data.filter_configs);
+        
+        publish_columns.forEach( col => {
+          if(Object.keys(filterConfigFilters).length > 0 && filterConfigFilters[col] != undefined && Object.keys(filterConfigFilters[col]).length > 0){
+            defultFilters[col] = filterConfigFilters[col];
```
Comment (jitendra-kumawat) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: move this if condition to small function

File: superset/assets/src/dashboard/reducers/getInitialState.js [hunk 6]
```diff
@@ -43,15 +43,56 @@ export default function(bootstrapData) {
...
-    );
-  } catch (e) {
-    //
+
+  const getFiltersFromFilterConfig = (filter_configs) => {
+    const filters = {};
+    filter_configs.forEach ( filter => {
+      const defaultValue = filter.hasOwnProperty("defaultValue") ? filter["defaultValue"] : "";
+      var values = defaultValue !=="" ? defaultValue.split(";") : []; 
+      if (values.length > 0) {
```
Comment (jitendra-kumawat) [inline superset/assets/src/dashboard/reducers/getInitialState.js]: add space between  defaultValue !==""

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 1 above)
Comment (ankursinghal2005) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: done

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 2 above)
Comment (ankursinghal2005) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: 1. done
2. As we discussed not needed...component behaviour is like this
3. Not required as per my view column is compulsory filed in chart

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 3 above)
Comment (ankursinghal2005) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: done

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 4 above)
Comment (ankursinghal2005) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: We don't have any constant file regarding all VIZ types

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 6 above)
Comment (ankursinghal2005) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: I covered in formatting code

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 5 above)
Comment (ankursinghal2005) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: done

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 2 above)
Comment (jitendra-kumawat) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: k

File: superset/assets/src/dashboard/reducers/getInitialState.js (hunk 4 above)
Comment (jitendra-kumawat) [reply superset/assets/src/dashboard/reducers/getInitialState.js]: k

**Pull request merged**